### DISCUSSION TITLE
Problem creating live sandbox??

### DISCUSSION THREAD

**Discussion by maxweld** [dotansimha/graphql-code-generator#7625] [Q&A]
Hi - I have been asked to raise an issue following some questions I put on chat. 
Now I am struggling with teh creation of a 'live sandpit'.
I am getting an 'unexpected error'. It looks like it is having trouble installing the version of the '@graphql-codegen/typescript' plugin that I am using (v^2.4.5) but it is not clear.
Sandbox is nifty-tamas-33dzx7
@charlypoly - Are you able to help?

**Comment by maxweld** [comment]
I have resolved the 'unexpected error' - issues with the dependencies.

It does not seem to run the codegen or generate new types

**Comment by maxweld** [comment]
I have made some progress, but it is still not generating types. Seems to be some problem with the schema, which was generated using Graphql-codegen/introspection

**Reply by charlypoly** [reply]
Hi @maxweld,

one thing that you can do, it setup this repository/fork locally and push changes to it so I can inspect it in codesandbox or pull it locally on my machine

**Reply by maxweld** [reply]
Hi @charlypoly - How do I do that? Sorry - but I am a bit new to all this.

**Reply by maxweld** [reply]
Hi @charlypoly - I have worked it out. setting it up at present. Will let you know when it is ready.

**Reply by maxweld** [reply]
Hi @charlypoly . 
I have managed to create a local repository based on the sandbox template. Sandbox was a bit funny, so this is preferred. It works in VS Code, but I am not sure how to get it running in the sandbox. I will try that again later. Anyway, thank you for the suggestion.

I think we could close this discussion now, but first I would like your advice on how to raise the issues I have. 

I am developing a protytype Vue app, but since I cannot decide on the best graphql client to use, I have two separate branches, one using the Apollo Client and the other using the URQL client. Both use Vue 3 with its new Composition API <script setup> syntax and access a Hasura graphql api which fronts a Postress DB. 

With the Apollo Client Branch, it is not clear if the @vue/apollo-composable package is mature, as it is still an alpha package (@^4.0.0-alpha16), and has been for months. Furthermore, from the Graphql-codegen website it is difficult to identify which of the 2 plugins should be used to generate the required Types and Queries/Mutations ( and what the codegen config should contain). **Are you able to advise or should I raise a new Discussion?**

With the Vue URQL Branch, I am more advanced. While I have a problem with Types, at this stage I dont think it is a codegen issue.

I need to decide between URQL or Apollo pretty soon, as it is doing my head in swapping between them. Could you suggest a preferance purely from a Codegen point of view?

**Reply by charlypoly** [reply]
@maxweld I'm glad to hear that you move forward with your setup!

Regarding your Apollo vs Vue question, from the codegen POV, we provide a plugin for both URQL and Apollo, however, the Apollo Composition plugin proposes better documentation: https://www.graphql-code-generator.com/plugins/typescript-vue-apollo

I have no idea of the current support status of Apollo for Vue.
I'll you make your own mind on the [URQL Comparaison page](https://formidable.com/open-source/urql/docs/comparison/#framework-bindings).

**Reply by maxweld** [reply]
Thanks @charlypoly. You say that the Apollo Composition plugin proposes better documentation, but the very first line on that page says to use a different plugin: 

> Instead of using this plugin, it is recommended that you use the [TypedDocumentNode plugin](https://www.graphql-code-generator.com/plugins/typed-document-node.md) together with the [Vue Apollo library](https://apollo.vuejs.org/).

That is part of my confusion. Why does it say that, and to what version of Vue does that apply?

**Reply by charlypoly** [reply]
Sorry for the confusion.

The TypeDocumentNode plugin is indeed recommended since it provides a better developer experience and lighter code generation.
Instead of generating hook functions, it generates typed document variable that you can pass to Vue Apollo's `useQuery()` function (works with Vue 4).
